packages/keycloak-user-management/src/components/Credentials/index.tsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import { Button, Form, Col, Card, Row, Input, Switch, notification } from 'antd';
+import { RouteComponentProps } from 'react-router';
+import { Store } from 'redux';
+import { connect } from 'react-redux';
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import { history } from '@onaio/connected-reducer-registry';
+import { HeaderBreadCrumb } from '../HeaderBreadCrumb';
+import {
+  fetchKeycloakUsers,
+  getAccessToken,
+  KeycloakUser,
+  makeKeycloakUsersSelector,
+  reducer as keycloakUsersReducer,
+  reducerName as keycloakUsersReducerName,
+} from '@opensrp/store';
+import { KeycloakService } from '@opensrp/keycloak-service';
+import '../../index.css';
+
+reducerRegistry.register(keycloakUsersReducerName, keycloakUsersReducer);
+
+/** inteface for route params */
+
+export interface RouteParams {
+  userId: string;
+}
+
+/** props for editing a user view */
+export interface Props {
+  accessToken: string;
+  fetchKeycloakUsersCreator: typeof fetchKeycloakUsers;
+  keycloakUser: KeycloakUser | null;
+  serviceClass: typeof KeycloakService;
+}
+
+/** interface for data fields for team's form */
+export interface UserCredentialsFormFields {
+  password: string;
+  confirm: string;
+  temporary: boolean;
+}
+
+/** type intersection for all types that pertain to the props */
+export type PropsTypes = Props & RouteComponentProps<RouteParams>;
+
+/** default props for editing user component */
+export const defaultProps: Partial<PropsTypes> = {
+  accessToken: 'hunter 2',
+  fetchKeycloakUsersCreator: fetchKeycloakUsers,
+  keycloakUser: null,
+  serviceClass: KeycloakService,
+};
+
+/** Handle form submission */
+export const submitForm = (values: UserCredentialsFormFields, props: PropsTypes): void => {
+  const { serviceClass, match, accessToken } = props;
+  const userId = match.params.userId;
+  const serve = new serviceClass(
+    accessToken,
+    `/users/${userId}/reset-password`,
+    'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage'
+  );
+  const { password, temporary } = values;
+  serve
+    .update({
+      temporary: temporary,
+      type: 'password',
+      value: password,
+    })
+    .then(() => {
+      history.push('/admin');
```

Comment:
- Must we use history.push?  If we are using a router package then lets use what it provides
- no magic strings please